src/Storage/Storage/Blob/Cmdlet/NewAzureStorageBlobSasToken.cs

```diff
@@ -141,7 +141,19 @@ public NewAzureStorageBlobSasTokenCommand(IStorageBlobManagement channel)
         /// </summary>
         [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
         public override void ExecuteCmdlet()
-        {
+        {           
+            // When the input context is Oauth bases, can't generate normal SAS, but UserDelegationSas
+            bool generateUserDelegationSas = false;
+            if (Channel != null && Channel.StorageContext != null && Channel.StorageContext.StorageAccount.Credentials.IsToken)
+            {
+                generateUserDelegationSas = true;
+                WriteVerbose("Will generate User Delegation SAS, since input Storage Context is OAuth based.");
```

Comment:
This should be included in the 'ShouldProcess' string as well

Reply:
fixed